src/components/Devices/Devices.tsx

```diff
@@ -103,45 +90,35 @@ function Devices() {
         };
       });
 
-    const getPublicKeyFak = async (existingPublicKeyFak) => {
-      if (existingPublicKeyFak !== 'null') return existingPublicKeyFak;
-      if (await isPassKeyAvailable()) {
-        const publicKey = await window.fastAuthController.getPublicKey();
-        return publicKey;
-      }
-      // Non WebAuthN supported browser
-      const keypair = await window.fastAuthController.getKey('oidc_keypair')
-        || await window.fastAuthController.getKey(window.fastAuthController.getAccountId());
-      return keypair.getPublicKey().toString();
-    };
-
     return controller.deleteDeviceCollections(list)
       .then(async () => {
         setisDeleted(false);
         setCollections(collections.filter((collection) => (!deleteCollections.includes(collection.id))));
         setDeleteCollections([]);
-        setIsAddingKey(true);
-
-        const email = decodeIfTruthy(searchParams.get('email'));
         const contract_id = decodeIfTruthy(searchParams.get('contract_id'));
-        const methodNames = decodeIfTruthy(searchParams.get('methodNames'));
-        const success_url = decodeIfTruthy(searchParams.get('success_url'));
-        const oidcToken = await controller.getUserOidcToken();
-
-        await onSignIn({
-          accessToken:      oidcToken,
-          publicKeyFak:     await getPublicKeyFak(publicKeyFak),
-          public_key_lak:   public_key_lak !== 'null' ? public_key_lak : decodeIfTruthy(searchParams.get('public_key')),
-          contract_id,
-          methodNames,
-          setStatusMessage: () => null,
-          success_url,
-          email,
-          searchParams,
-          navigate,
-          gateway:          success_url,
-        });
-        setIsAddingKey(false);
+
+        if (contract_id && public_key_lak) {
+          setIsAddingKey(true);
+
+          const email = decodeIfTruthy(searchParams.get('email'));
+          const methodNames = decodeIfTruthy(searchParams.get('methodNames'));
+          const success_url = decodeIfTruthy(searchParams.get('success_url'));
+          const oidcToken = await controller.getUserOidcToken();
+          await onSignIn({
+            accessToken:      oidcToken,
+            publicKeyFak,
```

Comment:
hmm after removing `getPublicKeyFak` have you tried to test create account, signin and multiple key flow on firefox?

Reply:
Yes it works on my side